core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt

```diff
@@ -731,6 +731,33 @@ object UniqueTriggerActivation {
                     true
                 }
             }
+
+            UniqueType.Eureka -> {
+                val tech = unique.params[1]
+                val amount = unique.params[0].toFloatOrNull()
+                if (amount == null || tech.isEmpty() || civInfo.tech.isResearched(tech)) return null
+                val scienceGain = (civInfo.tech.costOfTech(tech) * amount / 100).roundToInt()
+                if (scienceGain == 0) return null
+
+                return {
+                    if (civInfo.tech.techsInProgress[tech] == null) {
+                        civInfo.tech.techsInProgress.putIfAbsent(tech,scienceGain)}
```

Comment:
```suggestion
                        civInfo.tech.techsInProgress[tech] = scienceGain
                        }
```
We **know** it's absent at this point, so just a regular put